src/windows/wslaservice/exe/WSLASession.h

```diff
@@ -15,18 +15,22 @@ Module Name:
 #pragma once
 
 #include "wslaservice.h"
+#include "WSLAVirtualMachine.h"
 
 namespace wsl::windows::service::wsla {
 
 class DECLSPEC_UUID("4877FEFC-4977-4929-A958-9F36AA1892A4") WSLASession
     : public Microsoft::WRL::RuntimeClass<Microsoft::WRL::RuntimeClassFlags<Microsoft::WRL::ClassicCom>, IWSLASession, IFastRundown>
 {
 public:
-    WSLASession(const WSLA_SESSION_SETTINGS& Settings);
+    WSLASession(const WSLA_SESSION_SETTINGS& Settings, WSLAUserSessionImpl& userSessionImpl, const VIRTUAL_MACHINE_SETTINGS& VmSettings);
     IFACEMETHOD(GetDisplayName)(LPWSTR* DisplayName);
+    IFACEMETHOD(GetVirtualMachine)(IWSLAVirtualMachine** VirtualMachine);
 
 private:
-    std::wstring m_displayName;
+    WSLA_SESSION_SETTINGS m_sessionSettings;
+    WSLAUserSessionImpl& m_userSession;
+    Microsoft::WRL::ComPtr<WSLAVirtualMachine> m_virtualMachine;
```

Comment:
That might cause some issues in the tests, because we're keeping a COM reference to `m_virtualMachine`, which can prevent the service from stopping.

I'm wondering if we can solve this by just not making a ComPtr, and just using a "regular" instance, and calling `QueryInterface` instead of `CopyTo` in GetVirtualMachine. 

Let me know if that makes sense, i'm worried that if we check this in, the service failing to stop is going to cause a bit of pain

Reply:
I forget why we ended up making it a ComPtr in the first place, but makes sense.